pilot/pkg/networking/core/v1alpha3/networkfilter.go

```diff
@@ -102,11 +102,13 @@ func buildOutboundNetworkFiltersWithWeightedClusters(env *model.Environment, nod
 	}
 
 	for _, route := range routes {
-		clusterName := istio_route.GetDestinationCluster(route.Destination, push.ServiceByHostname[model.Hostname(route.Destination.Host)], port.Port)
-		clusterSpecifier.WeightedClusters.Clusters = append(clusterSpecifier.WeightedClusters.Clusters, &tcp_proxy.TcpProxy_WeightedCluster_ClusterWeight{
-			Name:   clusterName,
-			Weight: uint32(route.Weight),
-		})
+		if route.Weight > 0 {
```

Comment:
You can skip this if you want.. if weight is 0, envoy wont send any traffic. But it will make sure that the cluster is loaded..

Reply:
I think know your meaning. Just leave 0 weight cluster there for debugging.

Reply:
Seems we can not set weight=0

https://github.com/envoyproxy/envoy/blob/5f7c3de24f0d31515cd9a82aec482c3f225e9c8a/api/envoy/config/filter/network/tcp_proxy/v2/tcp_proxy.proto#L163